src/pkg/services/m365/sites_test.go

```diff
@@ -216,3 +222,146 @@ func (suite *siteUnitSuite) TestGetAllSites() {
 		})
 	}
 }
+
+func userIdentitySet(userID string, userEmail string) models.IdentitySetable {
+	user := models.NewIdentitySet()
+	userIdentity := models.NewUserIdentity()
+	userIdentity.SetId(ptr.To(userID))
+
+	if len(userEmail) > 0 {
+		userIdentity.SetAdditionalData(map[string]any{
+			"email": userEmail,
+		})
+	}
+
+	user.SetUser(userIdentity)
+
+	return user
+}
+
+func groupIdentitySet(groupID string, groupEmail string) models.IdentitySetable {
+	groupData := map[string]any{}
+	if len(groupEmail) > 0 {
+		groupData["email"] = groupEmail
+	}
+
+	if len(groupID) > 0 {
+		groupData["id"] = groupID
+	}
+
+	group := models.NewIdentitySet()
+	group.SetAdditionalData(map[string]any{"group": groupData})
+
+	return group
+}
+
+func dummySite(owner models.IdentitySetable) models.Siteable {
+	site := models.NewSite()
+	site.SetId(ptr.To("id"))
+	site.SetWebUrl(ptr.To("weburl"))
+	site.SetDisplayName(ptr.To("displayname"))
+
+	drive := models.NewDrive()
+	drive.SetOwner(owner)
+	site.SetDrive(drive)
+
+	return site
+}
+
+func (suite *siteUnitSuite) TestGetSites() {
+	table := []struct {
+		name       string
+		mock       func(context.Context) api.GetByIDer[models.Siteable]
+		expectErr  func(*testing.T, error)
```

Comment:
you can simplify this by using the assert comparison func type:
```suggestion
		expectErr  assert.ErrorAssertionFunc
```
Then on line 364 call:
```go
    test.expectErr(t, err, clues.ToCore(err))
```